zcash_client_sqlite/src/wallet/scanning.rs

```diff
@@ -715,8 +717,17 @@ pub(crate) fn update_chain_tip<P: consensus::Parameters>(
     params: &P,
     new_tip: BlockHeight,
 ) -> Result<(), SqliteClientError> {
+    // If the caller provided a chain tip that is before Sapling activation, do nothing.
+    let sapling_activation = match params.activation_height(NetworkUpgrade::Sapling) {
+        Some(h) => h,
+        None => return Ok(()),
+    };
+
```

Comment:
This doesn't directly check whether `new_tip` is before Sapling activation because it doesn't use `new_tip`, so the comment is confusing.

Reply:
This is a bug: the old code did check this condition, but in the refactor the comment was moved without fully moving the logic.